Formula/l/llama.cpp.rb

```diff
@@ -0,0 +1,21 @@
+class LlamaCpp < Formula
+  desc "LLM inference in C/C++"
+  homepage "https://github.com/ggerganov/llama.cpp"
+  # pull from git tag to get submodules
+  url "https://github.com/ggerganov/llama.cpp.git",
+  tag:      "b2950",
+  revision: "db10f01310beea8a1ef7798651b9d692fd1149d0"
+  license "MIT"
+
+  def install
+    system "make", "DLLAMA_FATAL_WARNINGS=ON", "DLLAMA_METAL_EMBED_LIBRARY=ON", "DLLAMA_CURL=ON"
+
+    bin.install "./main" => "llama-cli"
+    bin.install "./server" => "llama-server"
+  end
+
+  test do
+    llama_cli_command = "llama-cli -h"
```

Comment:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.

Reply:
Thanks for the advice @SMillerDev - I've added a more comprehensive test in there - such that it loads a model and then checks whether the model was loaded and then further offloaded too. Let me know if that works for you.